server/src/main/java/io/crate/metadata/functions/Signature.java

```diff
@@ -416,7 +430,20 @@ public void writeTo(StreamOutput out) throws IOException {
             TypeSignature.toStream(typeSignature, out);
         }
         TypeSignature.toStream(returnType, out);
-        out.writeVInt(EnumSets.packToInt(features));
+        if (out.getVersion().onOrAfter(Version.V_5_8_0)) {
+            out.writeVInt(EnumSets.packToInt(features));
+        } else {
+            // before 5.8.0, the ordinal for Scalar.Feature.NON_DETERMINISTIC was Scalar.Feature.DETERMINISTIC
+            if (features.contains(Scalar.Feature.NON_DETERMINISTIC)) {
+                var copy = new HashSet<>(features);
```

Comment:
double checking: it's because `features` is unmodifiable, right?

Reply:
Yes, that was the reason.